### PR TITLE
fixed some issues causing asm not to be compiled

### PR DESCRIPTION
**Pull request by jon-chuang**

### CODE REVIEW

**Comment by Pratyush** [comment]
Oops!

File: algebra-core/src/lib.rs [hunk 1]
```diff
@@ -6,13 +6,13 @@
 #![deny(unused_comparisons, bare_trait_objects, const_err, unused_must_use)]
 #![deny(unused_mut, unused_unsafe, private_in_public)]
 #![cfg_attr(
-    all(
+    not(all(
```
Comment (Pratyush) [inline algebra-core/src/lib.rs]: Hmm why do we want to negate here? `unsafe` is only required atm when all of these conditions are satisfied, right?

**Comment by Pratyush** [comment]
Ok I pushed a fix which should make it always compile. I also consolidated the conditional logic in `build.rs` to make it easier to audit. Thanks for bringing this to my attention!

**Comment by jon-chuang** [comment]
Ok, ran the bench, everything seems to be in order.

**Pull request merged**